File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder sb=new StringBuilder();
        for(String str: strs){
            sb.append(str);
            sb.append("_SSRK_");
        }
        return sb.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        String[] arrays_str=s.split("_SSRK_", -1);
        return Arrays.asList(arrays_str).subList(0, arrays_str.length-1);
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        res=[]
        for stri in strs:
            res.append(str(len(stri))+'#'+stri)

        return ''.join(res)
        

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        res=[]
        i=0

        while i<len(s):
            j=i
            while s[j]!='#': #getting the length of next string to read
                j+=1
            
            length_of_string=int(s[i:j])

            res.append(s[j+1: j+1+length_of_string])

            i = j + 1 + length_of_string
        
        return res


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))